src/index.js

```diff
@@ -62,25 +65,31 @@ var contourId = 0;
 
 function processPolygon(polygon, isSubject, depth, queue, bbox) {
   var i, len;
-  if (typeof polygon[0][0] === 'number') {
-    for (i = 0, len = polygon.length - 1; i < len; i++) {
-      processSegment(polygon[i], polygon[i + 1], isSubject, depth + 1, queue, bbox);
-    }
-  } else {
-    for (i = 0, len = polygon.length; i < len; i++) {
-      contourId++;
-      processPolygon(polygon[i], isSubject, contourId, queue, bbox);
-    }
+  for (i = 0, len = contourOrHole.length - 1; i < len; i++) {
+    processSegment(contourOrHole[i], contourOrHole[i + 1], isSubject, depth + 1, queue, bbox, isExteriorRing);
   }
 }
 
-
 function fillQueue(subject, clipping, sbbox, cbbox) {
   var eventQueue = new Queue(null, compareEvents);
-  contourId = 0;
-
-  processPolygon(subject,  true,  0, eventQueue, sbbox);
-  processPolygon(clipping, false, 0, eventQueue, cbbox);
+  subject.forEach(function (contour, counterIndex) {
```

Comment:
At the beginning of the `boolean` function you'll see I added a step to make sure that every input is an array of contours. So it could be an array with only a single contour, or it could be an array with multiple contours (eg a multipoly). My thinking in doing this was to give the array's a more consistent structure.
So for example an input should look some 
single poly = `[[contour, hole]]`
multipoly = `[[contour, hole], [contour, hole, hole]]`
This line of code loops through the available contours. Then the loop within it loops through the actual contour or holes.

Make sense? It seemed kinda logical to me but it may not be!